pkg/api/validation/validation.go

```diff
@@ -3198,10 +3198,21 @@ func ValidateConfigMap(cfg *api.ConfigMap) field.ErrorList {
 		for _, msg := range validation.IsConfigMapKey(key) {
 			allErrs = append(allErrs, field.Invalid(field.NewPath("data").Key(key), key, msg))
 		}
+		if _, isValue := cfg.BinaryData[key]; isValue {
+			allErrs = append(allErrs, field.Duplicate(field.NewPath("data").Key(key), key))
```

Comment:
This breaks old clients that try to patch a config map with a new data field that already exists in the binary field.